### ISSUE TITLE
Collectri issue with loops

### ISSUE THREAD

**Issue by dbdimitrov** [saezlab/omnipath#52]
Hi Denes,

Omnipath client version: '1.0.7'

I keep getting the same issue with collectri:

```
Downloading data from `https://omnipathdb.org/queries/enzsub?format=json`
Downloading data from `https://omnipathdb.org/queries/interactions?format=json`
Downloading data from `https://omnipathdb.org/queries/complexes?format=json`
Downloading data from `https://omnipathdb.org/queries/annotations?format=json`
Downloading data from `https://omnipathdb.org/queries/intercell?format=json`
Downloading data from `https://omnipathdb.org/about?format=text`
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
[/tmp/ipykernel_74765/1647182207.py](https://file+.vscode-resource.vscode-cdn.net/tmp/ipykernel_74765/1647182207.py) in 
----> 1 tf_net = dc.get_collectri(organism="mouse")

[~/anaconda3/envs/test-env/lib/python3.8/site-packages/decoupler/omnip.py](https://file+.vscode-resource.vscode-cdn.net/media/dbdimitrov/SSDDimitrov2/Repos/liana2_manuscript/notebooks/~/anaconda3/envs/test-env/lib/python3.8/site-packages/decoupler/omnip.py) in get_collectri(organism, split_complexes, **kwargs)
    424 
    425     # Load collectri
--> 426     ct = op.interactions.CollecTRI.get(genesymbols=True, organism=_organism, loops=True, **kwargs)
    427     if _organism == 'human':
    428         mirna = op.interactions.TFmiRNA.get(genesymbols=True, databases=['CollecTRI'], strict_evidences=True)

[~/anaconda3/envs/test-env/lib/python3.8/site-packages/omnipath/_core/requests/_utils.py](https://file+.vscode-resource.vscode-cdn.net/media/dbdimitrov/SSDDimitrov2/Repos/liana2_manuscript/notebooks/~/anaconda3/envs/test-env/lib/python3.8/site-packages/omnipath/_core/requests/_utils.py) in wrapper(wrapped, _instance, args, kwargs)
    112     @wrapt.decorator(adapter=wrapt.adapter_factory(argspec_factory))
    113     def wrapper(wrapped, _instance, args, kwargs):
--> 114         return wrapped(*args, **kwargs)
    115 
    116     from_class = hasattr(clazz, "get") and not hasattr(clazz.get, "__wrapped__")

[~/anaconda3/envs/test-env/lib/python3.8/site-packages/omnipath/_core/requests/_utils.py](https://file+.vscode-resource.vscode-cdn.net/media/dbdimitrov/SSDDimitrov2/Repos/liana2_manuscript/notebooks/~/anaconda3/envs/test-env/lib/python3.8/site-packages/omnipath/_core/requests/_utils.py) in _get_helper(cls, **kwargs)
     29         The result which depends the type of the request and the supplied parameters.
     30     """
---> 31     return cls()._get(**kwargs)
     32 
     33 

[~/anaconda3/envs/test-env/lib/python3.8/site-packages/omnipath/_core/requests/_request.py](https://file+.vscode-resource.vscode-cdn.net/media/dbdimitrov/SSDDimitrov2/Repos/liana2_manuscript/notebooks/~/anaconda3/envs/test-env/lib/python3.8/site-packages/omnipath/_core/requests/_request.py) in _get(self, **kwargs)
    106         kwargs = self._inject_fields(kwargs)
    107         kwargs, callback = self._convert_params(kwargs)
--> 108         kwargs = self._validate_params(kwargs)
    109         kwargs = self._finalize_params(kwargs)
    110         self._last_param["final"] = kwargs.copy()

[~/anaconda3/envs/test-env/lib/python3.8/site-packages/omnipath/_core/requests/_request.py](https://file+.vscode-resource.vscode-cdn.net/media/dbdimitrov/SSDDimitrov2/Repos/liana2_manuscript/notebooks/~/anaconda3/envs/test-env/lib/python3.8/site-packages/omnipath/_core/requests/_request.py) in _validate_params(self, params)
    193         for k, v in params.items():
    194             # first get the validator for the parameter, then validate
--> 195             res[self._query_type(k).param] = self._query_type(k)(v)
    196         return res
    197 

[~/anaconda3/envs/test-env/lib/python3.8/site-packages/omnipath/_core/query/_query.py](https://file+.vscode-resource.vscode-cdn.net/media/dbdimitrov/SSDDimitrov2/Repos/liana2_manuscript/notebooks/~/anaconda3/envs/test-env/lib/python3.8/site-packages/omnipath/_core/query/_query.py) in __call__(self, value)
    134     ) -> Optional[Set[str]]:
    135         """%(validate)s"""  # noqa: D401
--> 136         return self.value(value)
    137 
    138     @property

[~/anaconda3/envs/test-env/lib/python3.8/site-packages/omnipath/constants/_constants.py](https://file+.vscode-resource.vscode-cdn.net/media/dbdimitrov/SSDDimitrov2/Repos/liana2_manuscript/notebooks/~/anaconda3/envs/test-env/lib/python3.8/site-packages/omnipath/constants/_constants.py) in __call__(cls, *args, **kw)
     49                 f"without `__error_format__` class attribute."
     50             )
---> 51         return super().__call__(*args, **kw)
     52 
     53     def __new__(cls, clsname, superclasses, attributedict):  # noqa: D102

[~/anaconda3/envs/test-env/lib/python3.8/enum.py](https://file+.vscode-resource.vscode-cdn.net/media/dbdimitrov/SSDDimitrov2/Repos/liana2_manuscript/notebooks/~/anaconda3/envs/test-env/lib/python3.8/enum.py) in __call__(cls, value, names, module, qualname, type, start)
    337         """
    338         if names is None:  # simple value lookup
--> 339             return cls.__new__(cls, value)
    340         # otherwise, functional API: we're creating a new Enum type
    341         return cls._create_(

[~/anaconda3/envs/test-env/lib/python3.8/site-packages/omnipath/constants/_constants.py](https://file+.vscode-resource.vscode-cdn.net/media/dbdimitrov/SSDDimitrov2/Repos/liana2_manuscript/notebooks/~/anaconda3/envs/test-env/lib/python3.8/site-packages/omnipath/constants/_constants.py) in wrapper(*args, **kwargs)
     13             _cls, value, *_ = args
     14             e.args = (cls._format(value),)
---> 15             raise e
     16 
     17     if not issubclass(cls, ErrorFormatter):

[~/anaconda3/envs/test-env/lib/python3.8/site-packages/omnipath/constants/_constants.py](https://file+.vscode-resource.vscode-cdn.net/media/dbdimitrov/SSDDimitrov2/Repos/liana2_manuscript/notebooks/~/anaconda3/envs/test-env/lib/python3.8/site-packages/omnipath/constants/_constants.py) in wrapper(*args, **kwargs)
      9     def wrapper(*args, **kwargs) -> Enum:
     10         try:
---> 11             return fun(*args, **kwargs)
     12         except ValueError as e:
     13             _cls, value, *_ = args

[~/anaconda3/envs/test-env/lib/python3.8/enum.py](https://file+.vscode-resource.vscode-cdn.net/media/dbdimitrov/SSDDimitrov2/Repos/liana2_manuscript/notebooks/~/anaconda3/envs/test-env/lib/python3.8/enum.py) in __new__(cls, value)
    661                 ve_exc = ValueError("%r is not a valid %s" % (value, cls.__name__))
    662                 if result is None and exc is None:
--> 663                     raise ve_exc
    664                 elif exc is None:
    665                     exc = TypeError(

ValueError: Invalid value `loops` for `InteractionsQuery`. Valid options are: `['database', 'databases', 'dataset', 'datasets', 'directed', 'directeds', 'dorothea_level', 'dorothea_levels', 'dorothea_method', 'dorothea_methods', 'entity_type', 'entity_types', 'field', 'fields', 'format', 'formats', 'genesymbol', 'genesymbols', 'header', 'headers', 'license', 'licenses', 'limit', 'limits', 'organism', 'organisms', 'partner', 'partners', 'password', 'passwords', 'resource', 'resources', 'signed', 'signeds', 'source_target', 'source_targets', 'source', 'sources', 'target', 'targets', 'tfregulons_level', 'tfregulons_levels', 'tfregulons_method', 'tfregulons_methods', 'type', 'types']
```

**Comment by deeenes** [comment]
Unfortunately 1.0.7 doesn't tell much, because it was uploaded to PyPI in June, but later many bug fixes were added here, until I bumped the version to 1.0.8 and updated PyPI just yesterday. So first I'd try to install this latest version, as the ["loops" issue](https://github.com/saezlab/omnipath/issues/50) suggests your installed version is several months old.

**Comment by dbdimitrov** [comment]
Updating it to 1.0.8 fixed it, yeah I just noticed that 1.0.7 was fixed in my poetry env. I will close this issues. thanks Denes.
